### DISCUSSION TITLE
SSL handshake  failure

### DISCUSSION THREAD

**Discussion by nagu1137** [openssl/openssl#25995] [Q&A]
m_ssl = SSL_new(m_sslCtx);
SSL_set_fd(m_ssl, m_sock);
SSL_set_app_data(m_ssl, m_clsock);
we are getting the following error in the ssl_accept() method when we are trying to initate the connection.
Could you please let us know how to debug this, we are not getting what is the exact issue here. we are currently upgrading from 1.0.2 to 3.1.3

2024-11-14 01:51:04.245 - Info      - - Socket - /ftp - SSL handshake started.
2024-11-14 01:51:04.245 - Info      -  - Socket - /ftp - inside SSL_Info_Callback
2024-11-14 01:51:24.745 - Info      -- Socket - /ftp - inside SSL_Info_Callback
2024-11-14 01:51:24.745 - Info      - - Socket - /ftp - SSL error: before SSL initialization (in accept).

**Comment by Sashan** [comment]
Hello,

Do I understand right call to [`SSL_accept()`](https://docs.openssl.org/master/man3/SSL_accept/) fails? if so, what is the return value? Or does a code fails somewhere in `ssl_accelt()` function before even openssl gets called?

thanks for
clarification

**Comment by nagu1137** [comment]
Return value is -1 and as I printed error in the. Call back function error is
somewhere in accept.please look at my issue description

we are currently upgrading from 1.0.2 to 3.1.3

Nagaraju

**Comment by Sashan** [comment]
what `SSL_get_error()` reports? you should call this function after `SSL_accept()` fails. It should provide more detailed information. Seeing a return value `-1` from `SSL_accept()` can be anything from passing a file descriptor not associated with socket to client which hangs up unexpectedly.

**Comment by nagu1137** [comment]
Thanks for responding.

I'm on it...will.update here shortly
Nagaraju

**Comment by tom-cosgrove-arm** [comment]
This seems to be a question about the use of OpenSSL, rather than an issue (something not right with OpenSSL itself), and would therefore best be raised as a discussion

**Comment by nagu1137** [comment]
@Sashan  thanks for responding.  now we are able to work with implicit SSL enabling. Problem is the FTP client we were using windwos FTP which deosnt have SSL implict enabled.
now we used the WincSCP client with FTP SSL implict option then the server code is able to complete handshake.
Previosuly Looks like this should be the issue with Protocol version that is being negotiated

some times i get SSL protocol error-SSL_ERROR_SSL and some times i got SSL_ERROR_SYSCALL with SSL_get_error
